partiql-lang/src/main/kotlin/org/partiql/lang/eval/builtins/ScalarBuiltinsExt.kt

```diff
@@ -497,3 +500,30 @@ internal object ExprFunctionToString : ExprFunction {
         )
     }
 }
+
+/**
+ * Counts the number of characters in the specified string, where 'character' is defined as a single unicode code point.
+ *
+ * Same as CHARACTER_LENGTH, CHAR_LENGTH
+ */
+internal object ExprFunctionLength : ExprFunctionMeasure("length", StaticType.TEXT) {
+
+    override fun call(value: ExprValue): Int = characterLength(value)
+}
+
+/** text_replace(string, from, to) -- in [string], replaces each occurrence of [from] with [to].
+ */
+internal object ExprFunctionTextReplace : ExprFunction {
+    override val signature = FunctionSignature(
+        name = "text_replace",
+        requiredParameters = listOf(StaticType.TEXT, StaticType.TEXT, StaticType.TEXT),
+        returnType = StaticType.TEXT,
+    )
+
+    override fun callWithRequired(session: EvaluationSession, required: List<ExprValue>): ExprValue {
+        val string = required[0].stringValue()
+        val from = required[1].stringValue()
+        val to = required[2].stringValue()
+        return ExprValue.newString(string.replace(from, to))
```

Comment:
I don't believe it matters now, but we should consider writing string manipulation functions with the byte arrays / codepoints rather than the Kotlin builtins. but honestly if the result is the same then it doesn't matter.